Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -410,6 +411,38 @@ theorem append_castAdd_natAdd {f : Fin (m + n) → α} :
   unfold append addCases
   simp
 
+theorem append_injective_iff {α : Type*} {m n : ℕ} {x : Fin m → α} {y : Fin n → α} :
+    Function.Injective (Fin.append x y) ↔
+      Function.Injective x ∧ Function.Injective y ∧ Disjoint (Set.range x) (Set.range y) := by
```

Comment:
I think there's a shorter proof here which starts
```
theorem append_comp_finSumFinEquiv {α : Sort*} {m n : ℕ} {x : Fin m → α} {y : Fin n → α} :
    Fin.append x y ∘ finSumFinEquiv = Sum.elim x y := by
  ext (i | j) <;> simp

theorem append_injective_iff {α : Type*} {m n : ℕ} {x : Fin m → α} {y : Fin n → α} :
  Function.Injective (Fin.append x y) ↔
    Function.Injective x ∧ Function.Injective y ∧ Disjoint (Set.range x) (Set.range y) := by
  rw [← (finSumFinEquiv (m := m) (n := n)).injective_comp, append_comp_finSumFinEquiv]
```
and finishes with the lemma in #24899.

Though maybe this brings in an unwanted import for `finSumFinEquiv`

Reply:
That makes an import loop!